qiskit/primitives/base/base_sampler.py

```diff
@@ -77,22 +134,23 @@
 
 import warnings
 from abc import abstractmethod
-from collections.abc import Sequence
+from collections.abc import Iterable, Sequence
 from copy import copy
-from typing import Generic, TypeVar
+from typing import Generic, Optional, TypeVar
```

Comment:
`Optional` seems not to be used.
```suggestion
from typing import Generic, TypeVar
```